### ISSUE TITLE
lmdb: double free or corruption (fasttop)

### ISSUE THREAD

**Issue by anegrean** [man-group/ArcticDB#412]
Hi,

I tried to store 5 years worth of trades tick data from TSLA and INTC, and when I read back the data I get an error in python "double free or corruption (fasttop)". Core was dumped. I don't know if this is an issue with some upper data limit. I can share the stored data for testing and debug.

**Comment by qc00** [comment]
Hi, could you confirm the version and storage backend you're using?

It would also help if you could run the same action with `catchsegv` in front of the `python` command or upload the core dump.

**Comment by mehertz** [comment]
Hi - thanks for the bug report. Would you be ble to close & recreate this ticket using the bug template? Captures a lot of information for us and makes it easier for us to debug - for example would be good to understand exactly what sort of `read` you were doing.

**Comment by valankar** [comment]
I'm having the same issue, but with not much data. I'm using LMDB and the conda-forge package. Attached output.

```
python3 -c 'import sys; print("Python:", sys.version); import platform; print("OS:", platform.platform()); import arcticdb; print("ArcticDB:", arcticdb.__version__);'
Python: 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:58:44) [GCC 11.3.0]
OS: Linux-4.19.123-feral-x86_64-with-glibc2.24
ArcticDB: 1.4.1
```

[segv.txt](https://github.com/man-group/ArcticDB/files/11944689/segv.txt)

It was working fine for a while but then started. I'm using simple read() and append(). The library is created with dynamic_schema=True, but I have not modified any columns. The script seems to finish properly but get this error on exiting.

**Comment by mehertz** [comment]
Hi @valankar. Is this reproducable?

Are you able to share any of the data used?

**Comment by valankar** [comment]
> Hi @valankar. Is this reproducable?

Yes it continues to segfault as I keep appending data.

> Are you able to share any of the data used?

I've attached the db.

[accounts.arcticdb.tar.gz](https://github.com/man-group/ArcticDB/files/11948469/accounts.arcticdb.tar.gz)

**Comment by mehertz** [comment]
Thanks for the data sample @valankar. We'll take a look.

Re moving the database - this works for me:

```
In [5]: ac = Arctic('lmdb:///tmp/lmdbtest')

In [6]: lib = ac.create_library('test')

In [7]: !cp -r /tmp/lmdbtest /tmp/lmdbtest2

In [8]: ac = Arctic('lmdb:///tmp/lmdbtest2')

In [9]: ac.list_libraries()
Out[9]: ['test']
```

THat works how I'd expect. Do you have a little snippet to repro what you're talking about?

**Comment by valankar** [comment]
> THat works how I'd expect. Do you have a little snippet to repro what you're talking about?

This must have been an error on my part as I'm not able to reproduce it. Thanks.

**Comment by poodlewars** [comment]
hi @anegrean @valankar we fixed several similar LMDB issues in the `1.6.0` PyPi release. It would be worth retesting with that. Please let me know if you still encounter the issue and reopen this ticket. See issue #520 for details.
